pymongo/asynchronous/mongo_client.py

```diff
@@ -1453,13 +1453,6 @@ async def address(self) -> Optional[tuple[str, int]]:
                 'Cannot use "address" property when load balancing among'
                 ' mongoses, use "nodes" instead.'
             )
-        if topology_type not in (
-            TOPOLOGY_TYPE.ReplicaSetWithPrimary,
-            TOPOLOGY_TYPE.Single,
-            TOPOLOGY_TYPE.LoadBalanced,
-            TOPOLOGY_TYPE.Sharded,
-        ):
-            return None
```

Comment:
I believe returning None when the TOPOLOGY_TYPE is ReplicaSetNoPrimary was intentional and should be reinstated. This also warrants a not in the changelog and a `.. versionchanged::...`